### ISSUE TITLE
Does xmrig-nvidia support on ARM

### ISSUE THREAD

**Issue by rickyangtw** [xmrig/xmrig-nvidia#228]
xmrig-master can support ARM CPU.
Does xmrig-nvidia support on ARM?
When I try to build is now error message:
-------------------------------------------------
CMake Error at CMakeLists.txt:241 (add_executable):
  Cannot find source file:

    src/crypto/CryptoNight_arm.h
----------------------------------------------------

**Comment by yuzi-co** [comment]
are there any ARM systems that can use NVIDIA cards?

**Comment by muscovydux** [comment]
> 
> 
> are there any ARM systems that can use NVIDIA cards?
There are now, yes. NVIDIA released an ARM-based board with an integrated NVIDIA GPU. Any resolution to the issue above? Because I am experiencing it too.

**Comment by Spudz76** [comment]
Yeah, it appears we would need to build using [this Jetpack stuff](https://developer.nvidia.com/embedded/jetpack) and it's only supported on Debian, but that is also a decent sign that it might not be too tough to get it working.  CMake should understand cross-compiling just fine.  Also the devkit image runs kernel 4.9 which was really good for mining.  And the CUDA in the kit seems standard.

Of course devs have no hardware to really validate any guesses, but supposing a successful build is also successful at running, it could become alpha-supported and then tested by community.

**Comment by muscovydux** [comment]
Well I have the hardware so I would love to test any potential builds. I'm a bit out of my element on getting it to compile on my own though.

**Comment by Spudz76** [comment]
Which board do you have, it matters for which options in the SDK

Working on a true Ubuntu so the installer will work, it is a real pain in the ass about things (worse than CUDA installer!)

**Comment by muscovydux** [comment]
I have the Jetson-Nano which just came out. The official NVIDIA OS image is based on Ubuntu Desktop 18.04 and it comes with CUDA 10.0 already installed.

**Comment by Spudz76** [comment]
Awesome, thanks.  I never made it too far with the SDK installation, it's kind of a picky jerk about everything like most nVidia SDKs (OMG mint ~= ubuntu chill out and install anyway plz)

**Comment by sdoeringNew** [comment]
Just as information. With four threads and twelve blocks I get around 20H/s on my Jetson with the 5V/2A USB power connector. It takes ~8Wh in 90min.

**Comment by thinkpol2** [comment]
Thank you for the confirmation on the PR as well as the hashrate. When I tested original (ASIC) Cryptonight, I was seeing ~22h/s (barrel connector). Extrapolating from a similar-spec GPU, I'd speculated it could do ~44h/s but I believe it's limited by the 10 watt power budget.

**Comment by puccaso** [comment]
Hello all, 

I'm currently seeing about 100 h/s on a 8gb Raspberry pi 4b. Am i understanding correctly the Jetson only does 20 h/s? Should the Jetson not have a higher hashrate, or am I missing something? 

Puc
